nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ApplyTransformMarkLogic.java

```diff
@@ -96,28 +101,35 @@ public void init(ProcessorInitializationContext context) {
      */
     @Override
     protected QueryBatchListener buildQueryBatchListener(final ProcessContext context, final ProcessSession session) {
+
         ApplyTransformListener applyTransform = new ApplyTransformListener()
             .withApplyResult(
                 ApplyResultTypes.INGORE_STR.equals(context.getProperty(APPLY_RESULT_TYPE).getValue()) ? ApplyResult.IGNORE : ApplyResult.REPLACE
             )
             .withTransform(this.buildServerTransform(context))
             .onSuccess((batch) -> {
                 synchronized (session) {
+                    FlowFile incomingFlowFile = session.get();
+                    if (incomingFlowFile == null) {
+                        incomingFlowFile = session.create();
+                    }
                     for (String uri : batch.getItems()) {
-                        final FlowFile flowFile = session.create();
-                        session.putAttribute(flowFile, CoreAttributes.FILENAME.key(), uri);
-                        session.transfer(flowFile, SUCCESS);
+                        session.putAttribute(incomingFlowFile, CoreAttributes.FILENAME.key(), uri);
```

Comment:
This isn't quite right - per the JIRA requirement of "New FlowFiles should be created from the original FlowFile", we still want to create a new FlowFile for each item, but we need to do e.g. `FlowFile flowFile = session.create(incomingFlowFile);`. That means this method needs to be modified in the parent class so that `FlowFile incomingFlowFile` is passed in.